neo/Ledger/Blockchain.cs

```diff
@@ -610,6 +610,21 @@ private void Persist(Block block)
                 foreach (IPersistencePlugin plugin in Plugin.PersistencePlugins)
                     plugin.OnPersist(snapshot, all_application_executed);
                 snapshot.Commit();
+                foreach (IPersistencePlugin plugin in Plugin.PersistencePlugins)
+                {
+                    try
+                    {
+                        plugin.OnCommit(snapshot);
+                    }
+                    catch (Exception ex)
+                    {
+                        if (plugin.ShouldThrowExceptionFromCommit(ex))
+                        {
+                            lastException = ex;
+                        }
+                    }
+                }
+                if (lastException != null) throw lastException;
```

Comment:
What is the purpose of re-throwing the exception here?

Reply:
If this is the case, the plugin can actually write a log or print a warning, and there is no need to crash the node.

Reply:
Oh the reason for rethrowing after all plugins run is so that they all have an equal chance to commit. At this point can’t rollback the `snapshot.Commit()` anyway.

Reply:
> If this is the case, the plugin can actually write a log or print a warning, and there is no need to crash the node.

Author of the plugin could decide. If the plugin is for an exchange they may want it to crash anyway and have to resync the node. It could do more harm continuing to run, so it seems letting the plugin decide is better.

Reply:
If all they want to do is log, they can just return false from `plugin.ShouldThrowExceptionFromCommit(ex)` and log inside the method.